File: java/java/src/main/java/org/signal/zkgroup/profiles/ProfileKeyVersion.java
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.zkgroup.profiles;

import org.signal.zkgroup.InvalidInputException;
import org.signal.zkgroup.internal.ByteArray;
import java.io.UnsupportedEncodingException;

public final class ProfileKeyVersion extends ByteArray {

  public static final int SIZE = 64;

  public ProfileKeyVersion(byte[] contents) throws InvalidInputException {
    super(contents, SIZE);
  }

  public ProfileKeyVersion(String contents) throws InvalidInputException, UnsupportedEncodingException {
    super(contents.getBytes("UTF-8"), SIZE);
  }

  public String serialize() {
    try {
      return new String(contents, "UTF-8");
    } catch (UnsupportedEncodingException e) {
      throw new AssertionError();
    }
  }

}

File: swift/Sources/LibSignalClient/zkgroup/ProfileKeyVersion.swift
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import Foundation

public class ProfileKeyVersion: ByteArray, @unchecked Sendable {
    public static let SIZE: Int = 64

    public required init(contents: Data) throws {
        try super.init(newContents: contents, expectedLength: ProfileKeyVersion.SIZE)
    }
}
